src/response/hessian.jl

```diff
@@ -99,13 +104,20 @@ Return δψ where (Ω+K) δψ = rhs
         x .= pack(Pδψ)
     end
 
+    # Precompute IFFT of ψ
+    ψ_real = map(enumerate(basis.kpoints)) do (ik, kpt)
+        stack(map(eachcol(ψ[ik])) do ψnk
+            ifft(basis, kpt, ψnk)
+        end)
+    end
+
```

Comment:
the reason we didn't do it this way was because it kills the RAM for larger systems. So I'd rather not do this unless there's a very good reason to do it...

Reply:
If the time saving is significant we might want to add it as an option, but I'm not convinced it's worth the complexity. Your testcase is silicon unit cell? What kpoint sampling? What are the timings for a reasonably converged calculation (eg kpoint grid 8x8x8)?

Reply:
Here is a Silicon profile with a kgrid of 8x8x8 and refined Ecut of 120. The starting point was obtained from Ecut=90. Only entries with >= 1000 samples are included. Only the first two commits of this PR have been applied, **not** the IFFT caching.

https://gist.github.com/Technici4n/faf102bf9aa1f368ed859557a4f97254

You can see that `apply_K` (`100902` samples) is roughly 2x slower than `apply_Ω` (`53891` samples), and that roughly half of `apply_K` is the IFFTs of `ψ` (`25305` samples + `20430` samples).

In this example however, the memory usage of `ψ` is roughly 0.5 GB, and the memory usage of its IFFT is roughly 13 GB. So we should probably not go for this optimization. Overall, it can only save 20-25 % of the total running time.

Reply:
OK, can you remove the last optimization and add a comment detailing the tradeoff, and then we merge this?